array-filter/filter.js

```diff
@@ -0,0 +1,31 @@
+const filter = function (array, predicate) {
+  const output = [];
+  for (var i = 0; i < array.length; i++) {
+    if (predicate(array[i]) === true) {
```

Comment:
We don't necessarily want to check strict equality to `true`. `.filter()` works on the "truthiness" of the predicate's return value.

https://developer.mozilla.org/en-US/docs/Glossary/Falsy